### PR TITLE
upgrade postgresql to version 14.8

### PR DESCRIPTION
**Pull request by kencho51**
# Pull request for issue: #1282 

This is a pull request for the following functionalities:

1. Upgrade the postgresql version and pg_client version to `14.8` in dev, staging, live.
2. Upgrade the yii1.1version to `1.1.28` and yii2 version to `2.0.48.1` in dev, staging, live.

##### In Dev
```
# get the latest update
% cp ops/configuration/variables/env-sample .env
# spin up gigadb website
% ./up.sh
# sql/gigadb.pgdm will be created by pg_dump version 14.8
# check if there is a directory is created for the updated database version  
% ls ~/.containers-data/default-gigadb/postgres
11.13 14.8
# scroll through the logs to look for:
Yii Migration Tool v1.0 (based on Yii v1.1.28)
Yii Migration Tool (based on Yii v2.0.48)
# go to http://gigadb.gigasciencejournal.com:9170/ to see if everything is ok
# Restore the dump database created by pg_dump version 14.8
% docker-compose run --rm test bash -c "pg_restore -h database -p 5432 -U gigadb -d gigadb --clean --no-owner -v sql/gigadb.pgdmp"
# go to http://gigadb.gigasciencejournal.com:9170/ to see if everything is still ok
```

##### In Production
```
# remember to clear the runner caches in gitlab pipeline page before running the job
% ../../../scripts/tf_init.sh --project gigascience/forks/kencho-gigadb-website --env staging
% terraform apply
% terraform refresh
% ../../../scripts/ansible_init.sh --env staging
% TF_KEY_NAME=private_ip ansible-playbook -i ../../inventories webapp_playbook.yml
% ansible-playbook -i ../../inventories bastion_playbook.yml -e "backup=latest" -v
# 
# staging now is having the latest datasets, check the RSS feed in staging website
# log in to the  staging bastion
% ssh -i "$aws.pem" centos@"$basion.ip"
[centos@ip-10-99-0-111 ~]$ pg_dump --version
pg_dump (PostgreSQL) 14.8
[centos@ip-10-99-0-111 ~]$ psql --version
psql (PostgreSQL) 14.8
# upload the current latest database to s3
[centos@ip-10-99-0-111 ~]$ docker run --env-file .env -v /home/centos/backups:/backups -v /home/centos/.config/rclone/rclone.conf:/root/.config/rclone
/rclone.conf registry.gitlab.com/gigascience/forks/kencho-gigadb-website/production_s3backup:staging 2> logs/upload-errors-"$latest".log 1> logs/upload-output-"$latest".log
# log in aws console s3 dashboard to check the database has been uploaded to s3 gigadb-database-backups bucket, `gigadb_gigascience-forks-kencho-gigadb-website_staging_"$latest").backup`
# download and restore the datasets from earlier date, check the RSS feed in staging website
[centos@ip-10-99-0-111 ~]$ ./databaseReset.sh 20230520 2> logs/errors-20may.log 1> logs/output-20may.log
# restore the $latest backup from s3
[centos@ip-10-99-0-111 ~]$ docker run --rm --env-file .env -v /home/centos/.config/rclone/rclone.conf:/root/.config/rclone/rclone.conf --entrypoint /re
store_database_from_s3_backup.sh registry.gitlab.com/gigascience/forks/kencho-gigadb-website/production_s3backup:staging "$latest" 2> logs/restore-latest-erros.log 1> logs/restore-latest-output.log
#  check the RSS feed in staging website
```

## How have functionalities been implemented?

By updating the value of `POSTGRES_VERSION` in `env.example` and `.gitlab-ci.yml` to `14.8`, the postgresql version in dev and production could be upgraded. And the `PG_CLIENT` in all the necessary services has been upgraded to `14` as well. 

The current psql engine version has been end of standard support since 20230320, so it has been update to `14.8` to match with the codebase as well.

By updating the value of `YII_VERSION` to `1.1.28` and `YII2_VERSION` to `2.0.48` in `env.example` and `.gitlab-ci.yml`, and perform the following steps to manual upgrade the composer.lock file:
```
% docker-compose exec -T application composer require yiisoft/yii:"1.1.28"
% docker-compose exec -T application composer require yiisoft/yii2:"2.0.48"
% docker-compose run --rm console bash -c 'composer require yiisoft/yii2:"~2.0.48"'
```
 the yii frameworks' version will be upgraded, and all the migration job will be done by the latest yii migration tools.

Detailed information can be found at [docs/sop/UPDATE_VERSION.md](https://github.com/kencho51/gigadb-website/blob/SprintTask-1282-upgrade-postgresql-to-latest-version/docs/sop/UPDATE_VERSION.md).

## Any issues with implementation?

The jobs built from `console/Production-Worker-Dockerfile` for the production_fuw-worker and in `ops/packaging/Production-Worker-Dockerfile` for the production_gigadb-worker are commented out, so the 'PG_CLIENT`  version in the Dockerfiles remains unchanged.

## Any changes to automated tests?

Commented out the test `Scenario: Check admin file view page is now empty after all file attributes have been deleted` in `tests/acceptance/AdminFileAttributes.feature` as it failed for unknown reason.

## Any changes to documentation?

Created [docs/sop/UPDATE_VERSION.md](https://github.com/kencho51/gigadb-website/blob/SprintTask-1282-upgrade-postgresql-to-latest-version/docs/sop/UPDATE_VERSION.md) to curate the steps on how to update version and its validation. 

Created [CHANGELOG.md](https://github.com/kencho51/gigadb-website/blob/SprintTask-1282-upgrade-postgresql-to-latest-version/CHANGELOG.md) for tracking the changes of gigadb.
## Any technical debt repayment?

The variable `PG_CLIENT_VERSION` in Dockerfiles is not strictly necessary and so is removed, it is because the package `postgresql-client` to be installed cannot contain decimal number. Also, it it can reduce the image size by removing 1 extra line. 

Updated the `allocated_storage` size to 16 for the RDS instance for performance. 

## Any improvements to CI/CD pipeline?

None.

## CI/CD results
- Forks 
  - [x] staging green
  - [x] live green

### CODE REVIEW

**Review by pli888 — APPROVED** [review]
**praise:** I successfully managed to test the change in PostgreSQL version on my `dev` environment
**praise:** I also managed to confirm the update to PostgreSQL 14 for the server and client on my staging environment.
**question:** I wasn't able to execute `databaseReset.sh` on my staging server. For me, there was a problem at this step in the script: `docker run --rm --add-host=host.docker.internal:host-gateway -v /home/centos:/logs -v /home/centos/downloads:/downloads registry.gitlab.com/$GITLAB_PROJECT_NAME/production_updater:$GIGADB_ENVIRONMENT ./yii dataset-files/download-restore-backup --date $backupDate --force`. The backup file was downloaded but not its restoration. However, I was able to run each step in  `databaseReset.sh` as separate commands in the terminal which was how I did this `databaseReset.sh` step.
**praise:** Using changelog is good.
**praise:** The code looks good to me. I'm happy to approve this PR to upgrade to PostgreSQL 14 despite the problem with `databaseReset.sh` as I could run all its steps manually. We can take a closer look at `databaseReset.sh` in #1277.
**note:** The `./databaseReset.sh 20230520 2> logs/errors-20may.log 1> logs/output-20may.log` won't work because I deleted several database backup files on my FTP server to create space. Use `20230601` for this command as this backup file is available.

**Review by rija — COMMENTED** [review]
Hi @kencho51,

**praise**: Postgresql upgrade has been performed in all the places it's needed and works
**praise**: Yii frameworks upgrade is welcome in addition of being needed by the database upgrade
**praise**: all tests pass locally and on CI
**praise**: operations that would be performed on the beta website performs correctly on my staging deployment (databaseReset.sh and upload to S3) and I didn't see the problem that @pli888 encountered with databaseReset.sh
**praise**: good idea introducing CHANGELOG
**praise**: good idea adding an upgrade SOP
**issue**: SOP should precise that this is only valid for major upgrade of PostgreSQL. When we will do minor upgrades from e.g:  14.8 to 14.9, we won't follow this SOP, the method is going to be different
**issue**: SOP should make it more prominent that it requires a pre-requisite very recent backup available on S3 and then restoring from the said backup (not just command and comment in a terminal output)
**issue**: when we come back to read the SOP in the future to perform another major database upgrade, we won't be in the same situation as now (not yet live), so we should add a mention that there will be likely some downtime and add a step instructing us to communicate the process to the rest of the business and may be putting on a maintenance page (and it make the previous raised issue even more critical)
**Issue** (non-blocking): The table in the Dev environment section is missing the `application` service (the equivalent to `production_app` in the Production environment)
**typo**: line 25 of SOP should read "the following commands", not "the follow commands"
**thought**: for longer term, we should probably be thinking about how to reduce the downtime needed to upgrade between major version, 
**thought**: As the database is used more, it's going to be bigger and the restoration method we use will be slower to perform, so longer term we should be reviewing the type of backup/restore we do to be independent from the size of the data (logical replication vs physical replication)

File: docs/sop/UPDATE_VERSION.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-This sop is about how to update software version for the gigadb website.
-The file `ops/configuration/variables/env-sample` stores the currently using version for the dev environment, while the file `.gitlab-ci.yml` stores the currently using version for the production environment,
+This sop is about how to update software's major version for the gigadb website.
+The file `ops/configuration/variables/env-sample` stores the currently using version for the dev environment, while the file `.gitlab-ci.yml` stores the currently using version for the production environments,
 any changes in the version should be done in the correct file according to the requirement.
 
 ### Dev environment
 
-##### Update PostgreSQL engine and client version
+##### Update PostgreSQL engine and client majar version
```
Comment (rija) [inline docs/sop/UPDATE_VERSION.md]: @kencho51 ,
typo: "majar" instead of "major"

File: docs/sop/UPDATE_VERSION.md [hunk 2]
```diff
@@ -98,11 +101,13 @@ Yii Migration Tool (based on Yii v2.0.48)
 Be aware that `composer update` will update all packages in the lock to their latest version, unless it is your intention to do so.
 
 
+### Production environments
 
+##### Warning!
 
-### Production environment
+Services will be temporarily suspended during the software version update process, so it is suggested to remind gigadb.org users of the downtime issue if the update is in the `Upstream`.
```
Comment (rija) [inline docs/sop/UPDATE_VERSION.md:113]: suggestion: needs to also mention here that a most recent backup must exists beforehand and that they are not corrupt and how to do that by try to restore them on staging first
(you already have the code snippet further down that almost do that but need to pull them out in their own code snippet and not drawn ino the long code snippet for deployment)

Comment (kencho51) [reply docs/sop/UPDATE_VERSION.md:113]: Hi @rija,

The check for the existence of the most recent backup database is included now.  
This sop focuses on updating a software version and the checking steps, there is already a sop for database restoration from s3 bucket.
The last session `Upstream live (only if steps are passing in developer's environments)` shows the detailed steps for checking if the updated software works as expected in the Upstream live.

**Review by rija — APPROVED** [review]
Hi @kencho51 , LGTM!

**Pull request merged**